src/OrleansRuntime/MultiClusterNetwork/IGossipChannel.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Threading.Tasks;
+using Orleans.Runtime.Configuration;
+
+
+namespace Orleans.Runtime.MultiClusterNetwork
+{
+    /// <summary>
+    /// Interface for gossip channel.
+    /// 
+    /// A gossip channel stores multicluster data (configuration, gateways) and exchanges
+    /// this data with silos using a gossip-style communication, offering
+    /// two different methods (push or push-and-pull).
+    /// </summary>
+    public interface IGossipChannel
+    {
+        /// <summary>
+        /// Initialize the channel with given configuration.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        Task Initialize(GlobalConfiguration globalconfig, string connectionstring);
+
+        /// <summary>
+        /// A name for the channel.
+        /// </summary>
+        string Name { get; }
```

Comment:
Do we actually need `Name`? Wouldn't type be enough if there will be only one gossip channel of a type instantiated? In `AzureTableBasedGossipChannel` is it assigned the type name plus a static sequence number anyway.